pkg/api/v1alpha1/users.go

```diff
@@ -375,6 +410,19 @@ func (r *Router) updateUser(c *gin.Context) {
 		user.GithubUsername = null.StringFrom(req.GithubUsername)
 	}
 
+	wg := &sync.WaitGroup{}
+	updateNotificationPreferencesStatusCode := http.StatusOK
+	updateNotificationPreferencesError := error(nil)
+
+	wg.Add(1)
+
+	updateNotificationPreferences := func() {
+		defer wg.Done()
+
+		_, updateNotificationPreferencesStatusCode, updateNotificationPreferencesError = handleUpdateNotificationPreferencesRequests(c, r.DB, user, r.EventBus, req.NotificationPreferences)
+	}
+	go updateNotificationPreferences()
+
 	tx, err := r.DB.BeginTx(c.Request.Context(), nil)
```

Comment:
How does it work to pass `r.DB` to a separate async function above and then reuse it immediately after to start a transaction? Are the queries made in the async function then wrapped into the transaction as well? This feels like a race condition to me...